File: leetcode/easy/58.Length Of The Last Word/js/lengthOfTheLastWord.js
"use strict";

// Length of the Last Word

/**
 * 
 * @param {string} s
 * @return {number} 
 */
var lengthLastWord = function(s) {
    s = s.trim();
    let count = 0;
    let pointer = s.length - 1;
    while (pointer !== -1 && s[pointer] !== " ") {
        count++;
        pointer--;
    };
    return count;
};

// Testing
let s0 = "Hello World";
console.log(lengthLastWord(s0));

// Complexity Analysis:
// Time Complexity: O(N)
// Space Complexity: O(1)

File: leetcode/easy/58.Length Of The Last Word/python/lengthOfTheLastWord.py
"""Length of the Last Word"""

def length_last_word(s):
    s = s.strip()
    pointer = len(s) - 1
    count = 0
    while pointer != -1 and s[pointer] != " ":
        count += 1
        pointer -= 1
    return count

# Testing
s0 = "Hello World"
print(length_last_word(s0))

'''Complexity Analysis:
Time Complexity: O(N)
Space Complexity: O(1)'''